subsys/zigbee/lib/zigbee_fota/src/zigbee_fota.c

```diff
@@ -19,7 +19,7 @@ LOG_MODULE_REGISTER(zigbee_fota, CONFIG_ZIGBEE_FOTA_LOG_LEVEL);
 #define MANDATORY_HEADER_LEN   sizeof(zb_zcl_ota_upgrade_file_header_t)
 #define OPTIONAL_HEADER_LEN    sizeof(zb_zcl_ota_upgrade_file_header_optional_t)
 #define TOTAL_HEADER_LEN       (MANDATORY_HEADER_LEN + OPTIONAL_HEADER_LEN)
-#define SUBELEMENT_HEADER_SIZE 6
```

Comment:
This changes the value from 6 to 7, but the logic is not updated. What is the reasoning behind this change?

Reply:
Error during build samples using new version of toolchain which is more restrictive:
`/nrf/subsys/zigbee/lib/zigbee_fota/src/zigbee_fota.c:248:44: error: array subscript 'zb_zcl_ota_upgrade_sub_element_t {aka struct zb_zcl_ota_upgrade_sub_element_s}[0]' is partly outside array bounds of 'uint8_t[6]' {aka 'unsigned char[6]'} [-Werror=array-bounds]`

Reply:
I do not agree with this warning - the code never accesses `value` field, so it never does the out of bound access.
In my opinion - the fix is to define `zb_zcl_ota_upgrade_sub_element_hdr_t` that does not contain `value` field and use it instead of `zb_zcl_ota_upgrade_sub_element_t` for casting.